File: BankFrontend/src/app/services/cheque book/cheque-book.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { ChequeBook } from 'src/app/model/cheque_book';

@Injectable({
  providedIn: 'root'
})
export class ChequeBookService {
  private baseUrl: string = "http://localhost:11000/api/v1/bank/cheque-book";

  constructor(
    private http: HttpClient
  ) { }

  addChequeBook(chequeBook: ChequeBook): Observable<any> {
    return this.http.post(this.baseUrl, chequeBook);
  }

  getAllChequeBooks(): Observable<ChequeBook[]> {
    return this.http.get<ChequeBook[]>(this.baseUrl);
  }

  updateStatus(chequeBook: ChequeBook): Observable<any> {
    return this.http.put(this.baseUrl, chequeBook);
  }

  getByAccountNumber(accountNumber: number): Observable<ChequeBook> {
    return this.http.get<ChequeBook>(`${this.baseUrl}/${accountNumber}`);
  }
}

File: Backend/src/main/java/com/bank/demo/service/ChequeBookService.java
package com.bank.demo.service;

import java.util.List;

import org.springframework.stereotype.Component;

import com.bank.demo.model.ChequeBook;

@Component
public interface ChequeBookService {
	ChequeBook addChequeBook(ChequeBook chequeBook);
	List<ChequeBook> getAllChequeBooks();
	ChequeBook updateStatus(ChequeBook chequeBook);
}
